### ISSUE TITLE
[new DL][Accessibility] Focus state for Resource List is not visible in high contrast mode

### ISSUE THREAD

**Issue by melissaluu** [Shopify/polaris#3543]
## Issue summary
In high contrast mode, when an item in the Resource List is active/focused by keyboard navigation, it's not visible.

## Expected behavior
A visible focus/active state or outline is shown when a Resource List item is focused on.

## Actual behavior
![image](https://user-images.githubusercontent.com/6719231/96750973-9aec4980-139a-11eb-89f5-006b56be3416.png)

There's no visible highlight when the item is active/focused by keyboard navigation.

## Steps to reproduce the problem

1. Turn on high contrast mode
2. Go to a resource list within admin and keyboard navigate through the items to see if item is highlighted when it's active/focused
